sql/core/src/test/scala/org/apache/spark/sql/sources/BucketedReadSuite.scala

```diff
@@ -735,4 +735,54 @@ abstract class BucketedReadSuite extends QueryTest with SQLTestUtils {
         df1.groupBy("j").agg(max("k")))
     }
   }
+
+  //  A test with a partition where the number of files in the partition is
+  //  large. tests for the condition where the serialization of such a task may result in a stack
+  //  overflow if the files list is stored in a recursive data structure
+  //  This test is ignored because it takes long to run (~3 min)
+  ignore("SPARK-27100 stack overflow: read data with large partitions") {
+    val nCount = 12000
+    // reshuffle data so that many small files are created
+    val nShufflePartitions = 6000
+    // and with one table partition, should result in 6000 files in one partition
```

Comment:
Could you increase this like the following? The current value, `6000`, fails within IDE (IntelliJ) environment, but succeed SBT environment (`build/sbt "sql/testOnly *.BucketedReadWithoutHiveSupportSuite").
```
+    val nShufflePartitions = 10000
+    // and with one table partition, should result in 10000 files in one partition
```

Reply:
Done. So many things to remember to check :) Thank you for being so careful.

Reply:
Ur, @parthchandra . I gave you two lines including the comment.
Could you fix the comment, too?

Reply:
BTW, I didn't ask the `nCount`. Do you need to upgrade that for some reason?